src/Microsoft.Identity.Web/TokenCacheProviders/TokenCacheExtensions.cs

```diff
@@ -18,6 +20,9 @@ namespace Microsoft.Identity.Web
     /// </summary>
     public static class TokenCacheExtensions
     {
+        private static readonly Dictionary<MethodInfo, IServiceProvider> s_serviceProviderFromAction
```

Comment:
ConcurrentDictionary is thread safe, Dictionary is not.

Reply:
thanks. #resolve